packages/flutter/lib/src/widgets/animated_cross_fade.dart

```diff
@@ -0,0 +1,200 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/rendering.dart';
+import 'package:meta/meta.dart';
+
+import 'animated_size.dart';
+import 'basic.dart';
+import 'framework.dart';
+import 'transitions.dart';
+
+/// Specifies which of the children to show. See [AnimatedCrossFade].
+///
+/// The child that is shown will fade in, while the other will fade out.
+enum CrossFadeState {
+  /// Show the first child and hide the second.
+  showFirst,
+  /// Show the second child and hide the first.
+  showSecond
+}
+
+/// A widget that cross-fades between two children and animates itself between
+/// their sizes. The animation is controlled through the [crossFadeState]
+/// parameter. [firstCurve] and [secondCurve] represent the opacity curves of
+/// the two children, and will typically be [Interval] objects. Note that
```

Comment:
I would leave out the unsupported "will typically be..." statement. You'll need to fully explain how the intervals are typically used, perhaps below.